examples/autoqasm/2_Expressing_classical_control_flow.ipynb

```diff
@@ -0,0 +1,300 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "device = LocalSimulator()\n",
+    "result = device.run(conditioned_bell_program, shots=500).result()\n",
+    "counts = Counter(result.measurements[\"c\"])\n",
+    "print(\"measurement counts: \", counts)\n",
+    "\n",
+    "plt.bar(counts.keys(), counts.values())\n",
+    "plt.xlabel(\"bitstrings\")\n",
+    "plt.ylabel(\"counts\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "edee6e9c",
+   "metadata": {},
+   "source": [
+    "## For-loop"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ce1da10e",
+   "metadata": {},
+   "source": [
+    "In this example, we demonstrate by creating many Bell states in a quantum program. Instead of writing the preparation steps explicitly, we can use a for-loop to make the program simpler and more readable. \n",
+    "\n",
+    "The operations on classical variables (such as `i+1` on loop index `i` in this example) are only resolved at program execution time (i.e., during QPU or simulator executions). When the qubit index is determined by an expression, AutoQASM does not know how many qubits the program may need. In this case, we must define the qubit count using the keyword argument `num_qubits` when calling the quantum program. We can see that six qubits are declared in the line of generated OpenQASM script `qubit[6] __qubits__;`."
```

Comment:
Again, I think this is just distracting. Hopefully, it'll be less confusing when I move it to the decorator. I'll update these at that time

Reply:
If we are auto-detecting qubit counts for some case, but not the other, I think it's important to explain where is the boundary.

Reply:
unless users also need to declare qubit number, some explanation is needed